torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -176,7 +178,64 @@ struct GraphFuser {
     JIT_ASSERT(n->kind() == kFusionGroup);
     return *n->g(kSubgraph);
   }
+
+  void joinFusionGroups(Node *consumer_group, Node *producer_group) {
+    // Now we have two fusion groups!
```

Comment:
ASSERT it? :)

Reply:
I don't think it's necessary. Some functions I use inside `getSubgraph`, `mergeNodeIntoGroup` already do this check, so it will be caught.